apps/examples/testcase/le_tc/network/itc_net_inet.c

```diff
@@ -0,0 +1,237 @@
+*                     in the af address family into a character string*
+* @scenario          :passing hex value in inet_ntop
+* @apicovered        :inet_ntop()
+* @precondition      :None
+* @postcondition     :None
+*/
+static void itc_net_inet_ntop_p(void)
+{
+	struct in_addr in_addr;
+	char dst[INET_ADDRSTRLEN];
+	const char *ret;
+
+	in_addr.s_addr = 0x17071994;
+
+#ifdef CONFIG_NET_IPv4
+	ret = inet_ntop(AF_INET, &in_addr, dst, INET_ADDRSTRLEN);
+	TC_ASSERT_NEQ("inet_ntop", ret, NULL);
+#endif
+
+#ifdef CONFIG_NET_IPv6
+	ret = inet_ntop(AF_INET6, &in_addr, dst, INET_ADDRSTRLEN);
+	TC_ASSERT_NEQ("inet_ntop", ret, NULL);
+#endif
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase          :itc_net_inet_ntop_n
+* @brief             :function  converts  the network address structure src 
+*                     in the af address family into a character string*
+* @scenario          :passing invalid args to inet_ntop
+* @apicovered        :inet_ntop()
+* @precondition      :None
+* @postcondition     :None
+*/
+static void itc_net_inet_ntop_n(void)
+{
+	struct in_addr in_addr;
+	char dst[INET_ADDRSTRLEN];
+	const char *ret;
+
+	in_addr.s_addr = 0x17071994;
+
+#ifdef CONFIG_NET_IPv4
+	ret = inet_ntop(AF_INET, &in_addr, dst, 7);
+	TC_ASSERT_EQ("inet_ntop", ret, NULL);
+#endif
+#ifdef CONFIG_NET_IPv6
+	ret = inet_ntop(AF_INET6, &in_addr, dst, 7);
+	TC_ASSERT_EQ("inet_ntop", ret, NULL);
+#endif
+	ret = inet_ntop(33, &in_addr, dst, INET_ADDRSTRLEN);
+	TC_ASSERT_EQ("inet_ntop", ret, NULL);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         :itc_net_inet_pton_p
+* @brief            :function converts the character string src into a 
+*                    network address structure in the af address family
+* @scenario         :passing IPv4 addr in dot format and IPv6 in colon format
+* @apicovered       :inet_pton()
+* @precondition     :None
+* @postcondition    :None
+*/
+static void itc_net_inet_pton_p(void)
+{
+	struct sockaddr_in addr_inet;
+	int ret;
+
+#ifdef CONFIG_NET_IPv4
+	ret = inet_pton(AF_INET, "107.108.218.83", &(addr_inet.sin_addr));
+	TC_ASSERT_EQ("inet_pton", ret, 1);
+#endif
+#ifdef CONFIG_NET_IPv6
+	ret = inet_pton(AF_INET6, "0:0:0:0:0:0:0:1", &(addr_inet.sin_addr));
+	TC_ASSERT_EQ("inet_pton", ret, 1);
+#endif
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         :itc_net_inet_pton_n
+* @brief            :function converts the character string src into a 
+*                    network address structure in the af address family
+* @scenario         :passing invalid args to inet_pton
+* @apicovered       :inet_pton()
+* @precondition     :None
+* @postcondition    :None
+*/
+static void itc_net_inet_pton_n(void)
+{
+	struct sockaddr_in addr_inet;
+	int ret;
+
+#ifdef CONFIG_NET_IPv4
+	ret = inet_pton(AF_INET, "30051995", &(addr_inet.sin_addr));
+	TC_ASSERT_EQ("inet_pton", ret, 0);
+#endif
+#ifdef CONFIG_NET_IPv6
+	ret = inet_pton(AF_INET6, "0:0:0:0:0:0:0:1", &(addr_inet.sin_addr));
+	TC_ASSERT_EQ("inet_pton", ret, 1);
```

Comment:
isn't this case same as above test under CONFIG_NET_IPv6 ?
what s the difference?